result.h

```diff
@@ -844,6 +844,20 @@ struct Result {
         std::terminate();
     }
 
+    template<typename F = E>
+    typename std::enable_if<
+        !std::is_same<F, void>::value,
+        F
+    >::type
```

Comment:
That works. I talked about it with a friend and he made the exact same suggestion. I've made and pushed the changes to this PR.